### PR TITLE
new parser

### PR DESCRIPTION
**Pull request by nicolasparada**
Replacing the entire config parser with new code. This version does not use the [participle](https://github.com/alecthomas/participle) library.

Instead, for the classic mode, it splits by line, and detects commands (`@` prefix) and sections (`[NAMED]`). Then each property nested in a section is split by space. Simple.

It supports basic data types like string, integer, float and boolean. It does not try to parse regexes, dates, units, arrays, etc. Complex data types are left as string.

For the YAML parser, we use the [yaml](https://github.com/go-yaml/yaml) package with a custom unmarshaler to add support for sorted properties, so we don't loose track of the order of each property inside a plugin. This is mostly for testing so we can expect always the same order.

 - [x] classic config parser
 - [x] yaml config parser
 - [x] tests between them both
 - [x] support marshaling of multiline strings (`|-`)
 - [x] semantic schema validation
 - [x] json custom unmarshaler to preserve properties order too

### CODE REVIEW

File: testdata/multiline-value.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+customs:
+    - custom:
+        name: custom
+        multline: '"FOO\nBAR"'
```
Comment (nicolasparada) [inline testdata/multiline-value.yaml:4]: I would like if this were translated into:
```suggestion
        multline: -|
            FOO
            BAR
```

File: testdata/multiline-value.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+customs:
+    - custom:
+        name: custom
+        multiline: '"FOO\nBAR"'
```
[the remark this answers is not in the document]
Comment (nicolasparada) [reply testdata/multiline-value.yaml:4]: I'll look into it.

[the remark this answers is not in the document]
Comment (niedbalski) [reply testdata/multiline-value.yaml:4]: will you fix this @nicolasparada

[the remark this answers is not in the document]
Comment (nicolasparada) [reply testdata/multiline-value.yaml:4]: Yup

**Review by niedbalski — CHANGES_REQUESTED** [review]
Great work.

Looks generally good. I think the parsing schema for json might be a bit fragile. 

1. Did you remove the schema validator entirely from this parser?
2. Any chance you can re-instate the original tests? I am particularly interested
into booleans, multi-field, urls (with schema + port + uri) and the multiple-files ones.

File: testdata/multiline-value.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicolasparada) [reply testdata/multiline-value.yaml]: Done.

**Comment by nicolasparada** [comment]
1. I'm bringing the validator back.
2. I have already a good suite of tests, but sure I can add those :)

**Pull request merged**